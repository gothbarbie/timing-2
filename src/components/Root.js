import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'

import defaultStore from '../store/configureStore'

const Root = ({ store, children }) => (
  <Provider store={store ? store : defaultStore}>{children}</Provider>
)

Root.propTypes = {
  store: PropTypes.shape({}),
  children: PropTypes.node,
}

export default Root
