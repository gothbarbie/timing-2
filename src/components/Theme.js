import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
