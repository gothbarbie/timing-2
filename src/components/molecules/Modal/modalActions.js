import { HIDE_MODAL, SHOW_MODAL } from './modalTypes'

export const closeModal = () => ({
  type: HIDE_MODAL,
})

export const openModal = content => ({
  type: SHOW_MODAL,
  payload: content,
})
