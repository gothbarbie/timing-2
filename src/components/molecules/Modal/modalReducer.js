import { SHOW_MODAL, HIDE_MODAL } from './modalTypes'

const initialState = { content: 'addEvent', visible: false }

export default (state = initialState, action) => {
  switch (action.type) {
    case SHOW_MODAL:
      return { content: action.payload.content, visible: true }

    case HIDE_MODAL:
      return { ...state, visible: false }

    default:
      return state
  }
}
