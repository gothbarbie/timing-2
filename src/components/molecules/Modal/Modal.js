import React from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'

import { closeModal } from './modalActions'

import IconButton from 'generic/atoms/IconButton'

const Overlay = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.7);
  display: flex;
  align-items: center;
  justify-content: center;
`

const ModalStyle = styled.div`
  min-width: 300px;
  background: ${({ theme }) => theme.colors.background};
`

const ModalHeader = styled.header`
  background-color: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => theme.colors.white};
  font-size: 2.4rem;
  font-weight: 500;
  padding: 1rem 1rem 1rem 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
`

const ModalMain = styled.main`
  padding: 3rem;
`

class Modal extends React.PureComponent {
  onClickHandler = event => {
    const { closeModal } = this.props
    if (event.target.id === 'overlay') {
      closeModal()
    }
  }

  render() {
    const { closeModal, children, title, visible } = this.props

    return !visible ? null : (
      <Overlay id="overlay" onClick={this.onClickHandler}>
        <ModalStyle>
          <ModalHeader>
            {title}
            <IconButton
              color="white"
              hoverColor="gray"
              icon={['fas', 'times-circle']}
              onClick={closeModal}
              size="1x"
            />
          </ModalHeader>
          <ModalMain>{children}</ModalMain>
        </ModalStyle>
      </Overlay>
    )
  }
}

const mapStateToProps = ({ modal }) => {
  return {
    visible: modal.visible,
  }
}

const mapDispatchToProps = {
  closeModal,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Modal)
