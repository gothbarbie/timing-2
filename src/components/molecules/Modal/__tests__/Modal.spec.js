import React from 'react'
import { shallow } from 'enzyme'

import Root from '../../../Root'
import Theme from '../../../Theme'
import Modal from '../Modal'

describe('components/molecules/Modal', () => {
  it('renders', () => {
    const props = {
      closeModal: jest.fn(),
      title: 'Test Modal',
      visible: true,
    }

    const component = shallow(
      <Root>
        <Theme>
          <Modal {...props}>
            <div>Modal Content</div>
          </Modal>
        </Theme>
      </Root>
    )
      .dive()
      .dive()
      .dive()

    expect(component).toMatchSnapshot()
  })
})
