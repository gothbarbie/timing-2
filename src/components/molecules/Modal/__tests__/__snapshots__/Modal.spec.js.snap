// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/Modal renders 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
